src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzerTests.GuardedCallsTests.cs

```diff
@@ -0,0 +1,2595 @@
+
+class Test
+{
+    void M1(bool flag1, bool flag2)
+    {
+        [|M2()|];
+
+        if (OperatingSystemHelper.IsOSPlatformVersionAtLeast(""Windows"", 11))
+        {
+            M2();
+
+            if (flag1 || flag2)
+            {
+                M2();
+            }
+            else
+            {
+                M2();
+            }
+            
+            M2();
+        }
+
+        if (flag1 || flag2)
+        {
+            [|M2()|];
+        }
+        else
+        {
+            [|M2()|];
+        }
+    }
+
+    [SupportedOSPlatform(""Windows10.1.2.3"")]
+    void M2()
+    {
+    }
+}" + MockAttributesCsSource + MockOperatingSystemApiSource;
+            await VerifyAnalyzerAsyncCs(source);
+
+            var vbSource = @"
+Imports System.Runtime.Versioning
+Imports System
+
+Class Test
+    Private Sub M1(ByVal flag1 As Boolean, ByVal flag2 As Boolean)
+        [|M2()|]
+
+        If OperatingSystemHelper.IsOSPlatformVersionAtLeast(""Windows"", 11) Then
+            M2()
+
+            If flag1 OrElse flag2 Then
+                M2()
+            Else
+                M2()
+            End If
+
+            M2()
+        End If
+
+        If flag1 OrElse flag2 Then
+            [|M2()|]
+        Else
+            [|M2()|]
+        End If
+    End Sub
+
+    <SupportedOSPlatform(""Windows10.1.2.3"")>
+    Private Sub M2()
+    End Sub
+End Class
+" + MockRuntimeApiSourceVb + MockAttributesVbSource;
+            await VerifyAnalyzerAsyncVb(vbSource);
+        }
+
+        [Fact]
+        public async Task InterproceduralAnalysisTest()
+        {
+            var source = @"
+using System.Runtime.Versioning;
+using System;
+
+class Test
+{
+    void M1()
+    {
+        [|M2()|];
+
+        if (IsWindows11OrLater())
+        {
+            M2();    
+        }
+        else
+        {
+            [|M2()|]; 
+        }
+    }
+
+    [SupportedOSPlatform(""Windows10.1.2.3"")]
+    void M2()
+    {
+    }
+
+    bool IsWindows11OrLater()
+    {
+        return OperatingSystemHelper.IsOSPlatformVersionAtLeast(""Windows"",10,2,3,4);
+    }
+}" + MockAttributesCsSource + MockOperatingSystemApiSource;
+
+            await VerifyAnalyzerAsyncCs(source, "dotnet_code_quality.interprocedural_analysis_kind = ContextSensitive");
+        }
+
+        [Fact(Skip = "TODO: Analysis value not returned, needs to be fixed")]
+        public async Task InterproceduralAnalysisTest_LogicalOr()
+        {
+            var source = @"
+using System.Runtime.Versioning;
+using System;
+
+class Test
+{
+    void M1()
+    {
+        [|M2()|];
+
+        if (IsWindows11OrLater())
+        {
+            M2();    
+        }
+
+        [|M2()|]; 
+    }
+
+    [SupportedOSPlatform(""Windows10.1.2.3"")]
+    void M2()
+    {
+    }
+
+    bool IsWindows11OrLater()
+    {
+        return OperatingSystemHelper.IsOSPlatformVersionAtLeast(""Windows"",10,2,3,4) ||
+            OperatingSystemHelper.IsOSPlatformVersionAtLeast(""Windows"",11);
```

Comment:
This test disabled because interprocedural analysis not returning any result for combined platform checks with `AND or OR`  cc @mavasani